lib/engine.py

```diff
@@ -66,70 +69,31 @@ def get_connection(self):
     def add_to_table(self, data_source):
         """This function adds data to a table from one or more lines specified
         in engine.table.source."""
+        lines = gen_from_source(data_source)
+
         if self.table.columns[-1][1][0][:3] == "ct-":
             # cross-tab data
-
-            lines = gen_from_source(data_source)
             real_lines = []
```

Comment:
Excellent call moving this into it's own function. I think that the creation of `real_lines` should be moved inside the function as well.